### ISSUE TITLE
Turrets: Auto operate stop function after swap.

### ISSUE THREAD

**Issue by war3scv** [FakeFishGames/Barotrauma#11573]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Auto operate stop function after swap

### Reproduction steps

any turret with auto operate on, swap it at outpost, spawn crawler with console command.

### Bug prevalence

Happens every time I play

### Version

v1.0.8.0

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
Not actually a bug, because auto-operating is a property of a turret instance. So it's more an oversight.

**Comment by itchyOwl** [comment]
Actually, I'm not sure we should even implement this. If the sub builder decides that some turret should be automatically operated, that doesn't mean any other turret replacing it should too. Moreover, we'd have to copy all the properties related to auto operating, which is a bit tricky, because item swaps operate on ItemPrefab level and those properties are defined for the instances separately.

**Comment by war3scv** [comment]
> Actually, I'm not sure we should even implement this. If the sub builder decides that some turret should be automatically operated, that doesn't mean any other turret replacing it should too. Moreover, we'd have to copy all the properties related to auto operating, which is a bit tricky, because item swaps operate on ItemPrefab level and those properties are defined for the instances separately.

an effective way trun them on/off will be better by wire or something, I think we do? or we will just trun off allow swap, auto
turret fire on its own with or without periscope, I don't think anyone put periscope with it, so after swap you got a dead
turret. and probably workshop page comment bombardmentxD. and the main problem is the hardpoint if you want some auto 
turret and you want pay for it...well you can't.

**Comment by Wyald88** [comment]
My sub has 3 turrets and 3 hardpoints, I made a system where I can turn auto turret on and off meaning if you are playing solo and are lazy you can switch it on but if you are playing with friends you can keep it off or if you really like automode to stop the guns firing when its appropriate not to shoot. However, because of this issue my wiring was for nothing because now I either only have 3 guns set to auto whilst the others do nothing, giving no reason to buying them for solo lazy runs, or I load up all 6 guns and have 6 coilguns on the first stage. Not to mention they would be locked to only having coilguns, no options for lasers etc.

I really don't see why it wouldn't be presumed that a hardpoint set to auto means regardless of what you put on it it'll still be auto.

**Comment by allenaigit** [comment]
> Actually, I'm not sure we should even implement this. If the sub builder decides that some turret should be automatically operated, that doesn't mean any other turret replacing it should too. Moreover, we'd have to copy all the properties related to auto operating, which is a bit tricky, because item swaps operate on ItemPrefab level and those properties are defined for the instances separately.

I partially disagree with the first point: I believe most of the time auto operated turret reduces space usage and bot count of a sub, so when turrets are upgraded/swapped, these reason for auto operation still stands. If the devs still insist this is the desired behavior, then I strongly recommend clearly stating auto operated turret CANNOT be changed at all later in campaign in the sub editor, so design time choices can be made in advance.
With regard to difficulties migrating auto operation properties, I would recommend making it a switchable property that can be triggered by wire signals in connection panel. Even just the on/off switch with default values for delay/sway/aim precision should be sufficient for those unfortunate souls stuck with dead turret altogether.
Yes this is personal for I'm stuck half way into the campaign with 6 dead sentry turrets. It's frustrating since I cannot sell/destroy my current sub and replace it with updated design. :(

**Comment by itchyOwl** [comment]
I agree that if we decide implementing this, we should make it possible to toggle the auto-operating also in-game. But doing that, affects the balance and overall campaign difficulty. Currently auto-operating is not something we even use in the vanilla subs. And actually the feature was not intended to be used in the player subs in the first place. It's a leftover feature that we decided to keep, because it was (almost) functional. And it felt pity to cut it out just because _we_ didn't use it.

But let's see. Maybe we'll get back to this...

**Comment by Agent-Delta** [comment]
Hi, devs

I originally posted issue #11711 (since closed), concerning how Auto Operate could not be stopped by normal means. Although you ultimately resolved it with your own solution, I made [this mod](https://steamcommunity.com/sharedfiles/filedetails/?id=2961643561&tscn=1706691503) a couple weeks later. Just yesterday one of the subscribers commented asking about this very issue (11573), how trading out turrets/hardpoints "forgets" any previous settings for Auto Operate.

I had a feeling I'd find something like this here in the Issues section and sure enough I found this thread.

I totally understand how balance is a big concern here, but I really think (along with many others, apparently) that Auto Operate should be more accessible and consistent across the board.

First of all, I think it's a pretty safe bet that anybody using AO will just always want it to persist in that slot, regardless of what they have installed there. Second, the modding community for this game has become so prolific and accessible that a huge percentage of the playerbase is basically setting their own balance now. The fact that AO and my mod exist together means that anyone can use it; now it's only a matter of how frustrating (or simple) it is to work with. If players want a more vanilla submarine experience, they won't use Auto Operate.

Is this making any sense?
Thanks for all the work you have continued to put into this great game.
